determined-ai/determined
# chore: Move Agents into their own context, introduce Loadable abstraction

## Description
This is a PR @trentwatt and I put together. It does two things:

1. Split out Agents from the `Store` context. This is how we want to go about breaking up the Store context so that we can eventually switch to a more flexible way of managing global data.
2. Introduces a Loadable data type to manage loading states. Currently we're pretty bad at handling loading states and often just temporarily display incorrect data while loading. This will make it easier to implement loading states going forward and see where they're missing now.

![Screen Shot 2022-10-28 at 14 50 23](https://user-images.githubusercontent.com/1002726/198719157-df00a8ad-625b-422f-9c60-44e0d32ea007.png)
![Screen Shot 2022-10-28 at 15 09 42](https://user-images.githubusercontent.com/1002726/198719159-0d2127a6-880b-4861-9778-a38da26423ac.png)


## Test Plan
1. Visit the cluster overview page, confirm it matches latest-master (there are some features here I don't see in latest master if people have ideas on how to test them).
2. Confirm the navbar matches latest-master
4. Confirm the resource pool detail page matches latest-master
5. Replace each instance of `getAgents`, etc with `NotLoaded` and confirm the loading states display correctly.


## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
